bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -201,7 +201,12 @@ public void stateUpdated(Item item, State state) {
                 event.item = EnrichedItemDTOMapper.map(item, false, null, null);
 
                 // adjust the state according to the widget type
-                event.item.state = itemUIRegistry.getState(w).toFullString();
```

Comment:
With this change we break the API of the sitemap REST interface. The `event.item.state` is now not a widget specific conversion any more but the raw state.
How about turning this around and provide the raw state in `event.state` (in case it is different to `event.item.state` and leave the widget specific state on the item?

Reply:
Yes you're right but sitemap SSE is very probably only used in Basic UI and the windows app. So I will break only the windows app.
I think it would be confusing to put/keep the widget state in the item section and put the item state in the widget/main section.
Additionally the sitemap REST API provides the same item structure and in this case we get the item state. So it would be confusing for a developer to have 2 different behaviors. And my intention was to add the widget state to rhe REST sitemap API in a second time and this time without breaking the API.

I think it is reasonable to break the API as it will only impact an app not very used and still in a build state.
But if the backward compatibility is the first priority, yes I can exchange the two fields and we can choose the confusing solution.

Reply:
I agree with you that overwriting `event.item.state` with a widget specific state feels wrong and confusing in the first place. And also "mixing" in the REST interface is not good either.
However we do not know if and how other solutions do actually use the REST interface. Basic UI and the windows app are  only the consumers we know of. There might be others depending on the interface.
@maggu2810, @kaikreuzer & @sjka what is your take on this?

Reply:
> Yes you're right but sitemap SSE is very probably only used in Basic UI and the windows app. 

It's also used in my [ESH Grafana Javascript library](https://github.com/wborn/smarthome-grafana).

Reply:
And I hope you are not against my improvment ?!

Reply:
The proposed changes are probably the best way to solve the issues in all UIs without having to deal with legacy data in the future.

Is there some versioning on the interface that can be used to determine which fields should be used? That way it would be possible to consume the old/new API with the same code. The Windows App may also need this when it keeps supporting older OH2 versions. Of course I can also apply the same kind of change in my library that you've applied in Basic UI. 

It will probably cost people some time to figure out why it no longer works and what they have to do to fix it. Announcements and release notes may help with that.

Reply:
> @maggu2810, @kaikreuzer & @sjka what is your take on this?

I would prefer useful names instead of keeping backward compatibility.
A consumer of different version should check the used ESH version.